build-docker-image/action.yml

```diff
@@ -117,6 +122,54 @@ runs:
       env:
         DOCKER_BUILD_RECORD_UPLOAD: false
 
+    - name: Set up Docker Buildx
+      if: ${{ inputs.max-cache == 'true' }}
+      uses: docker/setup-buildx-action@v3
+
+    - name: Build docker image using max cache
+      uses: docker/build-push-action@v6
+      if: ${{ ( inputs.reuse-cache == 'true' && inputs.max-cache == 'true' ) }}
+      with:
+        tags: |
+          ${{ env.DOCKER_REPOSITORY }}:${{ env.IMAGE_TAG }}
+          ${{ env.DOCKER_REPOSITORY }}:${{ env.BRANCH_TAG }}
+        push: false
+        load: true
+        cache-to: |
+          type=registry,ref=${{ env.DOCKER_REPOSITORY }}:${{ env.IMAGE_TAG }}-buildcache,mode=max
+          type=registry,ref=${{ env.DOCKER_REPOSITORY }}:${{ env.BRANCH_TAG }}-buildcache,mode=max
+        cache-from: |
+          type=registry,ref=${{ env.DOCKER_REPOSITORY }}:main-buildcache
+          type=registry,ref=${{ env.DOCKER_REPOSITORY }}:${{ env.IMAGE_TAG }}-buildcache
```

Comment:
This should never be used as the tag is unique

Reply:
removed